Cmdline/Main.cs

```diff
@@ -174,6 +176,43 @@ public static int Main(string[] args)
 
             #endregion
 
+            try
+            {
+                log.InfoFormat("About to run action {0}",cmdline.action);
+                return RunAction(cmdline, options, args, user, manager);
+            }
+            catch (RegistryInUseKraken kraken)
+            {
+                log.Info("Registry in use detected");
+                user.RaiseMessage(kraken.ToString());
+
+                // By forcing the current instance to be null, we
+                // prevent the finally block from trying to do clean-up
+                // on a locked instance, which ironically throws another
+                // lock exception.
+                manager.CurrentInstance = null;
+                return Exit.ERROR;
+            }
+            finally
+            {
+                log.Info("Freeing resources");
+
+                // Release the registry lock file if possible.
+                if (manager.CurrentInstance != null)
+                {
+                    manager.CurrentInstance.RegistryManager.Dispose();
```

Comment:
If manager.CurrentInstance.RegistryManager is null, this will throw a NullReferenceException. Take a look [here](https://msdn.microsoft.com/en-us/library/ty67wk28.aspx) for an elegant solution to these kinds of issues.